### PR TITLE
config: move all opts items to opts module

### PR DESCRIPTION
**Pull request by wusyong**
There are some opt items are located in servoshell instead of config crate. This makes it difficult to refactor winit API mentioned in #34522. We should properly add them to `Opts` already.

---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors

- [x] These changes do not require tests because these are existing opts. The build should be able to be built though.

### CODE REVIEW

File: components/config/opts.rs [hunk 1]
```diff
@@ -135,6 +135,21 @@ pub struct Opts {
...
+    pub clean_shutdown: bool,
+
+    /// Device pixels per px.
+    pub device_pixel_ratio: Option<f32>,
+
+    /// Set custom user agent string (or ios / android / desktop for platform default).
+    pub user_agent: Option<String>,
+
+    /// Start with a custom URL to load.
+    pub url: Option<String>,
```
Comment (mrobinson) [inline components/config/opts.rs:152]: I don't think we should moving in the opposite direction: moving these to servoshell as command-line options are something that the embedder should handle and not Servo itself.

**Review by mrobinson — CHANGES_REQUESTED** [review]

File: components/config/opts.rs (hunk 1 above)
Comment (wusyong) [reply components/config/opts.rs:152]: But again some opt fields like `initial_window_size` also relate to embedder. And `pref` and `prefs-file` are defined in servoshell, but they are used in [prefs::register_user_prefs](https://github.com/servo/servo/blob/main/ports/servoshell/prefs.rs#L30). All these configurations are scattered around which I struggled to refactor servoshell properly.

Comment (mrobinson) [reply components/config/opts.rs:152]: I imagine what should happen is that `Opts` and be called `CommandLineOptions` and it should just adjust preferences. Servo shouldn't know about `CommandLineOptions` at all.

Comment (mrobinson) [reply components/config/opts.rs:152]: I agree that having some command-line options only in `servoshell` isn't great though! They should all be together, but command-line option handling should be an embedder concern.

Comment (wusyong) [reply components/config/opts.rs:152]: I see. I'll think how to overcome it in servoshell

**Pull request closed without merging**